cmd/notation/internal/sharedutils/inspect.go

```diff
@@ -0,0 +1,125 @@
+// Copyright The Notary Project Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sharedutils
+
+import (
+	"crypto/sha256"
+	b64 "encoding/base64"
+	"encoding/hex"
+	"fmt"
+	"github.com/notaryproject/notation-core-go/signature"
+	"github.com/notaryproject/notation/internal/cmd"
+	"github.com/notaryproject/notation/internal/tree"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"os"
+	"strings"
+	"time"
+)
+
+type InspectOutput struct {
+	MediaType  string `json:"mediaType"`
+	Signatures []SignatureOutput
+}
+
+type SignatureOutput struct {
+	MediaType             string              `json:"mediaType"`
+	Digest                string              `json:"digest"`
+	SignatureAlgorithm    string              `json:"signatureAlgorithm"`
+	SignedAttributes      map[string]string   `json:"signedAttributes"`
+	UserDefinedAttributes map[string]string   `json:"userDefinedAttributes"`
+	UnsignedAttributes    map[string]string   `json:"unsignedAttributes"`
+	Certificates          []certificateOutput `json:"certificates"`
+	SignedArtifact        ocispec.Descriptor  `json:"signedArtifact"`
+}
+
+type certificateOutput struct {
+	SHA256Fingerprint string `json:"SHA256Fingerprint"`
+	IssuedTo          string `json:"issuedTo"`
+	IssuedBy          string `json:"issuedBy"`
+	Expiry            string `json:"expiry"`
+}
+
+func GetSignedAttributes(outputFormat string, envContent *signature.EnvelopeContent) map[string]string {
```

Comment:
do we need all this functions to be public?

Reply:
No, reduced the scope.